### ISSUE TITLE
openmm.OpenMMException: Particle coordinate is NaN.

### ISSUE THREAD

**Issue by Kash-Pandz** [openmm/openmm#4587]
Hi,

I am running NVT equilibration and the following error is raised:

openmm.OpenMMException: Particle coordinate is NaN.

My system has been minimised properly with the potential energy converging. 

I know this issue has been flagged before due to OpenMM and CUDA versions, but I am still unsure as to how to fix the error. I am running simulations on four T4 GPUs using OpenMM 8.1.1 and CUDA 11.7.0. 

Any guidance is much appreciated.

**Comment by peastman** [comment]
See [this FAQ entry](https://github.com/openmm/openmm/wiki/Frequently-Asked-Questions#nan).  It has lots of discussion about things that can cause this problem and how to debug it.

**Comment by Kash-Pandz** [comment]
I've figured out what the issue was. It was the CustomExternalForce I was using was causing the issue. I removed it and and the NVT equilibration again and it worked fine.

I just wanted to check I'm applying the CustomExternalForce to the right set of atoms (i.e. protein heavy atoms only)?

    # Initialise the CustomExternalForce for positional restraints
    restraint_force = CustomExternalForce("k * periodicdistance(x, y, z, x0, y0, z0)^2")
    # Convert force constant to right units
    force = restraint_force * kilocalories_per_mole / angstroms**2
    # Add global paramter for force constant
    restraint_force.addGlobalParameter('k', 100.0*kilojoules_per_mole/nanometer)
    # Add per-particle parameters for reference positions
    restraint_force.addPerParticleParameter('x0')
    restraint_force.addPerParticleParameter('y0')
    restraint_force.addPerParticleParameter('z0')


    # Select heavy atoms of protein  
    for atom in pdbfile.topology.atoms():
        if atom.residue.name != 'HOH' and atom.element.symbol != 'H':
            restraint_force.addParticle(atom.index, pdbfile.positions[atom.index])

Thank you

**Comment by peastman** [comment]
If you have any ions, that will also apply it to them, which probably isn't what you want.  It might be safest to select just the particular chains you want.  For example,

```python
chains = list(pdbfile.topology.chains())
for atom in chains[0].atoms():
   ...
```

**Comment by Kash-Pandz** [comment]
Thank you for your help. Once I select the heavy atoms do I have to add the force to the system right? (as shown in this snippet)

    chains_list = list(pdbfile.topology.chains())
    chains = list(pdbfile.topology.chains())
    for chain_index in [0, 1]:
        for atom in chains[chain_index].atoms():
            if atom.element.symbol != 'H':
               restraint_force.addParticle(atom.index, pdbfile.positions[atom.index])

    system.addForce(restraint_force)

**Comment by peastman** [comment]
Correct.  The simulation only knows about forces that have been added to the system.
